### DISCUSSION TITLE
The column 'id' was specified multiple times for 'temp_table'

### DISCUSSION THREAD

**Discussion by wildfiregt** [yajra/laravel-datatables#2531] [Q&A]
### Summary of problem or feature request

There seems to be an issue when tables have same primary column names. First page loads but not the other pages.

### Code snippet of problem

Let's say we have 2 tables with this architecture :

`users (id, name)
comments (id, text, user_id)`

I try to eager load comments with users using eloquent : 

`$comments = Comment::query()->with('users');`

Logging the output gives the expected result 

`[
    {
        "id": 1,
        "text": "Comment",
        "id_user": 1,
        "user" : {
            "id" : 2,
            "name" : "user 2"
        }
    }
]`

So before using Yajra\Datatables, everything seems fine.

Then i call Yajra datatable like this : 
`return Datatables::of($comments)->make(true);`

First issue : when printing json returned by Yajra, the id of Comment is messed up and takes "user" id.
First page loads, when changing the page, i get the following error 
`The column 'id' was specified multiple times for 'temp_table'`

So there's an issue when tables have the same column name. When I changed one of those "id" to another name, the problem is gone. But that's not an ideal solution. Can't change column names of every database.

It is possible to change the request to raw sql, but that's not ideal either. Would be nice to solve this using default Eloquent models

### System details

- DBMS : MS SQL Server 2012
- Operating System : Windows 8.1
- PHP Version : 7.3.10
- Laravel Version : 5.5
- Laravel-Datatables Version : 9.0

**Comment by yajra** [comment]
You need to specify a select when eager loading a relationship.

```php
$comments = Comment::query()->select('comments.*')->with('users');
```
